Index: src/com/company/leetcode/problem/JumpGame.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/com/company/leetcode/problem/JumpGame.java	(date 1605617209621)
+++ src/com/company/leetcode/problem/JumpGame.java	(date 1605617209621)
@@ -0,0 +1,30 @@
+package com.company.leetcode.problem;
+
+import java.util.Arrays;
+
+public class JumpGame {
+    public static void main(String[] args) {
+        canJump(new int[]{3,2,1,0,4});
+    }
+
+
+    public static boolean canJump(int[] nums) {
+
+        if(nums.length <= 1)
+            return true;
+
+        int max = nums[0]; //max stands for the largest index that can be reached.
+
+        for(int i=0; i<nums.length; i++){
+            //if not enough to go to next
+            if(max <= i && nums[i] == 0) return false;
+            //update max
+            if(i + nums[i] > max) max = i + nums[i];
+            //max is enough to reach the end
+            if(max >= nums.length-1) return true;
+        }
+
+        return false;
+    }
+
+}
Index: src/com/company/study/section/section2/새로운게임.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/com/company/study/section/section2/새로운게임.java	(date 1605624710090)
+++ src/com/company/study/section/section2/새로운게임.java	(date 1605624710090)
@@ -0,0 +1,7 @@
+package com.company.study.section.section2;
+
+public class 새로운게임 {
+    public static void main(String[] args) {
+
+    }
+}
Index: src/com/company/leetcode/problem/RotateImage.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/com/company/leetcode/problem/RotateImage.java	(date 1605793940953)
+++ src/com/company/leetcode/problem/RotateImage.java	(date 1605793940953)
@@ -0,0 +1,30 @@
+package com.company.leetcode.problem;
+
+import java.util.Arrays;
+
+public class RotateImage {
+    public static void main(String[] args) {
+        int[][] matrix = new int[][]{{1,2,3}, {4,5,6}, {7,8,9}};
+        int[][] matrix2 = new int[][]{{5,1,9,11}, {2,4,8,10}, {13,3,6,7}, {15,14,12,16}};
+        rotate(matrix);
+        rotate(matrix2);
+    }
+
+
+    public static void rotate(int[][] matrix) {
+        int[][] result = new int[matrix.length][matrix.length];
+        for (int i=0; i<matrix.length; i++) {
+            for (int j=0; j<matrix.length; j++) {
+                result[j][i] = matrix[matrix.length-1-i][j];
+            }
+        }
+
+        for (int i=0; i<matrix.length; i++) {
+            for (int j=0; j<matrix.length; j++) {
+                matrix[i][j] = result[i][j];
+            }
+        }
+
+        System.out.println(Arrays.deepToString(matrix));
+    }
+}
